packages/esm-patient-forms-app/src/htmlformentry/html-form-entry-wrapper.component.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import styles from './html-form-entry-wrapper.scss';
+
+interface HtmlFormEntryWrapperProps {
+  closeWorkspace: () => void;
+  src: string;
+}
+
+const HtmlFormEntryWrapper: React.FC<HtmlFormEntryWrapperProps> = ({ closeWorkspace, src }) => {
+  useEffect(() => {
+    const iframe = document.querySelector('iframe');
+    iframe.addEventListener('load', () => {
+      // hide the header and breadcrumbs
+      const dashboard = iframe.contentDocument;
+      dashboard.querySelector('header')?.remove();
+      dashboard.querySelector('.patient-header')?.remove();
+      dashboard.querySelector('#breadcrumbs')?.remove();
+      iframe.style.display = 'block';
+    });
+
+    // register a listener to close the workspace when HFE sends a "close-workspace" message
+    window.addEventListener('message', (event) => {
+      if (event.data === 'close-workspace') {
+        closeWorkspace();
+      }
```

Comment:
I've made a small change to HTML Form Entry UI so that the if the "returnUrl" message is in the format: 'post-message:some-message' it posts a message "some-message" instead of attempting to redirect to "returnUrl'

Also: I see there is closeWorkspace and a closeWorkspaceWithSavedChanges.  Is there an explanation of the different between these two (and which one should I be calling here)

Reply:
The difference is basically that `closeWorkspace()` will notify the user if they close the workspace with unsaved changes whereas `closeWorkspaceWithSavedChanges()` doesn't check if there are unsaved changes.